src/DI/ContainerBuilder.php

```diff
@@ -57,8 +60,9 @@ public function addDefinition($name, ServiceDefinition $definition = NULL)
 	{
 		if (!is_string($name) || !$name) { // builder is not ready for falsy names such as '0'
 			throw new Nette\InvalidArgumentException(sprintf('Service name must be a non-empty string, %s given.', gettype($name)));
-
-		} elseif (isset($this->definitions[$name])) {
+		}
+		$name = isset($this->aliases[$name]) ? $this->aliases[$name] : $name;
```

Comment:
Maybe addDefinition() should be supposed to create only "original" services, so what about to throw exception when `if (isset($this->definitions[$name]) || isset($this->aliases[$name]))`?